shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/Alert.kt

```diff
@@ -203,4 +205,64 @@ data class Alert(
     fun anyInformedEntity(predicate: (InformedEntity) -> Boolean) = informedEntity.any(predicate)
 
     fun matchingEntities(predicate: (InformedEntity) -> Boolean) = informedEntity.filter(predicate)
+
+    companion object {
+        /**
+         * Filter the list of alerts to only the ones that match the given filter parameters and
+         * affect boarding.
+         */
+        fun filter(
+            alerts: List<Alert>,
+            directionId: Int?,
+            routeIds: List<String>,
+            stopIds: Set<String>
+        ): List<Alert> {
+            return applicableAlerts(
+                routes = routeIds,
+                stopIds = stopIds,
+                directionId = directionId,
+                alerts = alerts
+            )
+        }
+
+        /**
+         * Gets the alerts of the first stop that is downstream of the target stop which has alerts.
+         *
+         * @param alerts: The full list of alerts
```

Comment:
Do you do these comments fully by hand? I've used tools for Java that help generate similar inline documentation, it might be helpful to see if there's anything that would work for `shared`

Reply:
I did! I don't always add the `params`, it felt relevant here since I felt like the params could use a little more explanation. I see [a few different KDoc generator plugins to try ](https://plugins.jetbrains.com/search?excludeTags=internal&search=kdoc)